packages/cosmic-proto/README.md

```diff
@@ -1,9 +1,298 @@
+        - Cosmos, CosmWasm, and IBC
+            - [CosmWasm](#cosmwasm-messages)
+            - [IBC](#ibc-messages)
+            - [Cosmos](#cosmos-messages)
+- [Wallets and Signers](#connecting-with-wallets-and-signing-messages)
+    - [Stargate Client](#initializing-the-stargate-client)
+    - [Creating Signers](#creating-signers)
+    - [Broadcasting Messages](#broadcasting-messages)
+- [Advanced Usage](#advanced-usage)
+- [Developing](#developing)
+- [Credits](#credits)
+
+## Usage
+### RPC Clients
+
+```js
+import { agoric } from '@agoric/cosmic-proto';
+
+const { createRPCQueryClient } = agoric.ClientFactory; 
+const client = await createRPCQueryClient({ rpcEndpoint: RPC_ENDPOINT });
+
+// now you can query the cosmos modules
+const balance = await client.cosmos.bank.v1beta1
+    .allBalances({ address: 'agoric1addresshere' });
+
+// you can also query the agoric modules
+const balances = await client.agoric.exchange.v1beta1
+    .exchangeBalances()
+```
+
+### Composing Messages
+
+Import the `agoric` object from `@agoric/cosmic-proto`. 
+
+```js
+import { agoric } from '@agoric/cosmic-proto';
+
+const {
+    createSpotLimitOrder,
+    createSpotMarketOrder,
+    deposit
+} = agoric.exchange.v1beta1.MessageComposer.withTypeUrl;
+```
+
+#### CosmWasm Messages
+
+```js
+import { cosmwasm } from "@agoric/cosmic-proto";
+
+const {
+    clearAdmin,
+    executeContract,
+    instantiateContract,
+    migrateContract,
+    storeCode,
+    updateAdmin
+} = cosmwasm.wasm.v1.MessageComposer.withTypeUrl;
+```
+
+#### IBC Messages
+
+```js
+import { ibc } from '@agoric/cosmic-proto';
+
+const {
+    transfer
+} = ibc.applications.transfer.v1.MessageComposer.withTypeUrl
+```
+
+#### Cosmos Messages
+
+```js
+import { cosmos } from '@agoric/cosmic-proto';
+
+const {
+    fundCommunityPool,
+    setWithdrawAddress,
+    withdrawDelegatorReward,
+    withdrawValidatorCommission
+} = cosmos.distribution.v1beta1.MessageComposer.fromPartial;
+
+const {
+    multiSend,
+    send
+} = cosmos.bank.v1beta1.MessageComposer.fromPartial;
+
+const {
+    beginRedelegate,
+    createValidator,
+    delegate,
+    editValidator,
+    undelegate
+} = cosmos.staking.v1beta1.MessageComposer.fromPartial;
+
+const {
+    deposit,
+    submitProposal,
+    vote,
+    voteWeighted
+} = cosmos.gov.v1beta1.MessageComposer.fromPartial;
+```
+
+## Connecting with Wallets and Signing Messages
+
+⚡️ For web interfaces, we recommend using [cosmos-kit](https://github.com/cosmology-tech/cosmos-kit). Continue below to see how to manually construct signers and clients.
+
+Here are the docs on [creating signers](https://github.com/cosmology-tech/cosmos-kit/tree/main/packages/react#signing-clients) in cosmos-kit that can be used with Keplr and other wallets.
+
+### Initializing the Stargate Client
+
+Use `getSigningagoricClient` to get your `SigningStargateClient`, with the proto/amino messages full-loaded. No need to manually add amino types, just require and initialize the client:
```

Comment:
Is the capitalisation of `agoric` incorrect in the Telescope config, or is this just a documentation bug.  Either way, it would be better to make it `getSigningAgoricClient` since it is part of the public interface.

Reply:
bad find-replace in the Markdown. I'll fix